### PR TITLE
[lzav] Add new port

### PR DESCRIPTION
**Pull request by chausner**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/lzav/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,11 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO avaneev/lzav
+    REF ${VERSION}
+    SHA512 f3882fa13a63aadec89b627a941c1bd6c35384aeb79baad61fe558c92b0481103a7b78bbfd880ce72d030a9e0fb7293fed789a5ca33476d303849376da5d9e96
+    HEAD_REF master
+)
+
+file(COPY "${SOURCE_PATH}/lzav.h" DESTINATION "${CURRENT_PACKAGES_DIR}/include/lzav")
```
Comment (chausner) [inline ports/lzav/portfile.cmake:9]: Should we install the header into its own folder like I do here? The official README provides a usage example with just `#include <lzav.h>`.

Comment (LilyWangLL) [reply ports/lzav/portfile.cmake:9]: I think so.

**Review by LilyWangLL — DISMISSED** [review]

File: ports/lzav/portfile.cmake (hunk 1 above)
Comment (dg0yt) [reply ports/lzav/portfile.cmake:9]: Does the usage heuristic give the right instructions to use `#include <lzav.h>`?

Comment (chausner) [reply ports/lzav/portfile.cmake:9]: It suggests `#include <lzav/lzav.h>` which should be correct, no?

Comment (dg0yt) [reply ports/lzav/portfile.cmake:9]: > It suggests `#include <lzav/lzav.h>` which should be correct, no?

The suggestion works, but it is significantly different from upstream's `#include "lzav.h"`.  
(Yes, `"`. I assume they don't expect the single file to be passed through a complex package manager.)  
But there will be hardly any package which has one of the include statements but not the file...

Comment (chausner) [reply ports/lzav/portfile.cmake:9]: Sorry, I edited my comment above. The usage hint actually does not provide any example how to include the header. All it says is:

> lzav is header-only and can be used from CMake via:
>
> find_path(LZAV_INCLUDE_DIRS "lzav/lzav.h")
> target_include_directories(main PRIVATE ${LZAV_INCLUDE_DIRS})

So would you suggest to keep the port as it is or should I change it to install the header directly under `include`?

**Comment by BillyONeal** [comment]
I want to give upstream an opportunity to comment on the correct usage: https://github.com/avaneev/lzav/issues/5

**Comment by chausner** [comment]
> I want to give upstream an opportunity to comment on the correct usage: [avaneev/lzav#5](https://github.com/avaneev/lzav/issues/5)

This is the answer given by the maintainers:

> I can't give a definitive suggestion here, as this depends on vcpkg design guidelines (e.g., requirement of inclusion of upstream readme and license files), which I do not know. For sure, lzav is going to remain a single header file project - that's one of its advantages over the competition.

What do you suggest on how to proceed?

**Comment by BillyONeal** [comment]
> What do you suggest on how to proceed?

I tried clarifying the problem for upstream. I think we should give them a few more days to think about it and if they don't say otherwise we should go with `#include <lzav.h>` (that is, install into `${CURRENT_PACKAGES_DIR}/include`) since that's closest to `#include "lzav.h"` upstream says while still being packagable.

**Comment by chausner** [comment]
Since the discussion didn't yield any definitive conclusion, I changed the port to install the header directly into `include`, as you suggested @BillyONeal.

**Pull request merged**